app/dashboard/views.py

```diff
@@ -1444,3 +1446,95 @@ def new_bounty(request):
         update=bounty_params,
     )
     return TemplateResponse(request, 'bounty/new.html', params)
+
+
+@csrf_exempt
+def change_bounty(request, bounty_id):
+    """Change bounty."""
+    if request.body:
+        user = request.user if request.user.is_authenticated else None
+
+        if not user:
+            return JsonResponse(
+                {'error': _('You must be authenticated via github to use this feature!')},
+                status=401)
+
+        try:
+            params = json.loads(request.body)
+        except Exception:
+            return JsonResponse({'error': 'Invalid JSON.'}, status=405)
+
+        try:
+            bounty_id = int(bounty_id)
+            bounty = Bounty.objects.get(pk=bounty_id)
+        except Bounty.DoesNotExist:
+            return JsonResponse({'error': _('Bounty doesn\'t exist!')}, status=404)
+
+        # TODO(review): check bounty status?
+        is_funder = bounty.is_funder(user.username.lower()) if user else False
+        if not is_funder:
+            return JsonResponse({
+                'error': _('You must be funder to change the bounty.')
+            }, status=401)
+
+        item = int(params.get('expirationTimeDelta', 0))
+        if item:
+            deadline = round(time.time() + item)
+            bounty.expires_date = timezone.make_aware(
+                timezone.datetime.fromtimestamp(deadline),
+                timezone=UTC)
+
+        item = params.get('attached_job_description', '')
```

Comment:
i wonder if we could use django forms for any of this repetitive code.... cc @mbeacom